lib/model/CHierarchicalResultsNormalizer.cc

```diff
@@ -123,16 +123,30 @@ void CHierarchicalResultsNormalizer::visit(const CHierarchicalResults& /*results
                        ? 0.0
                        : maths::CTools::anomalyScore(node.probability());
 
+    // Construct a unique identifier for the current partition. This will be used to store the maximum scores per
+    // partition
+    const std::string partitionName = (node.s_Spec.s_PartitionFieldName)
+                                          ? *(node.s_Spec.s_PartitionFieldName)
+                                          : "";
+    const std::string partitionValue = (node.s_Spec.s_PartitionFieldValue)
+                                           ? *(node.s_Spec.s_PartitionFieldValue)
+                                           : "";
+    const std::string personName =
+        (node.s_Spec.s_PersonFieldName) ? *(node.s_Spec.s_PersonFieldName) : "";
+    const std::string personValue =
+        (node.s_Spec.s_PersonFieldValue) ? *(node.s_Spec.s_PersonFieldValue) : "";
```

Comment:
Can we use a static const empty string here to avoid copying strings (maybe you can factor out as something like a `const std::string& dereferenceOrEmpty(const std::string&))`.